src/conversion.inl

```diff
@@ -47,49 +47,66 @@ inline constexpr Mat4 toMatrix4(const Quat& q) noexcept
 
 inline Quat toQuaternion(const Mat3& m) noexcept
 {
-    Quat ret;
+    Quat q;
     const f32 trace = m.trace();
 
     if (trace > .0f)
     {
         f32 tmp = sqrtf(trace + 1.f);
 
-        ret.s = tmp * .5f;
+        q.s = tmp * .5f;
 
         tmp = .5f / tmp;
 
-        ret.v.x = (m[5] - m[7]) * tmp;
-        ret.v.y = (m[6] - m[2]) * tmp;
-        ret.v.z = (m[1] - m[3]) * tmp;
+        q.v.x = (m.e[5] - m.e[7]) * tmp;
+        q.v.y = (m.e[6] - m.e[2]) * tmp;
+        q.v.z = (m.e[1] - m.e[3]) * tmp;
     }
 
     else
     {
         u8 i = 0;
-        if (m[4] > m[0])      i = 1u;
-        if (m[8] > m[4u * i]) i = 2u;
+        if (m.e[4] > m.e[0])      i = 1u;
+        if (m.e[8] > m.e[4u * i]) i = 2u;
 
         const u8 next[3]    = {1u, 2u, 0u},
                  j          = next[i],
                  k          = next[j];
 
-        f32 tmp = sqrtf((m(i, j) - (m(j, j) + m(k, k))) + 1.f);
+        f32 tmp = sqrtf(m.c[j].e[i] - (m.c[j].e[j] + m.c[k].e[k]) + 1.f);
 
-        ret[i] = tmp * .5f;
+        q.e[i] = tmp * .5f;
 
         if (tmp != .0f)
             tmp = .5f / tmp;
 
-        ret.v.z = (m(k, j) - m(j, k)) * tmp;
-        ret[j]  = (m(j, i) + m(i, j)) * tmp;
-        ret[k]  = (m(k, i) + m(i, k)) * tmp;
+        q.v.z  = (m.c[j].e[k] - m.c[k].e[j]) * tmp;
+        q.e[j] = (m.c[i].e[j] + m.c[j].e[i]) * tmp;
+        q.e[k] = (m.c[i].e[k] + m.c[k].e[i]) * tmp;
     }
 
-    return ret;
+    return q;
 }
 
 
 inline Quat toQuaternion(const Mat4& m) noexcept
 {
     return toQuaternion(toMatrix3(m));
 }
+
+
+inline SplitTransform toSplitTransform(const Transform& transfo) noexcept
+{
+    return {toQuaternion(transfo.rotation()), transfo.translation(), transfo.scaling()};
+}
+
+
+inline Transform toTransform(const SplitTransform& transfo) noexcept
+{
+    return
+    {
+        transfo.position,
+
```

Comment:
Rotation and scale look like missing?

Reply:
Wups, I probably messed up a cut/paste. Fixed!